tasks/lib/DocumentParser.js

```diff
@@ -81,8 +90,14 @@ class DocumentParser {
         }
       }
       if (line.trim().endsWith('-->')) {
-        this.inComment = false;
-        this.currentTag = this.nextTag(i);
+        if (line.trim().endsWith('--->')) {
+          this.inMetadata = false;
+          this.document.metadata = JSON.parse(this.metadata);
+          this.currentTag = this.nextTag(i);
```

Comment:
I'm not sure we should record the nextTag here. This is only relevant if we start a new section. Better to remove this line.